sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -123,6 +123,8 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper {
         // Signed values in [-8388608, 8388607] and unsigned values in [0, 16777215],
         // both of them fit IntegerType
         Some(IntegerType)
+      case Types.REAL | Types.FLOAT =>
+        if (md.build().getBoolean("isSigned")) Some(FloatType) else Some(DoubleType)
```

Comment:
Just a question. We are safe even if MySQL community remove UNSIGNED FLOAT , right?
- https://dev.mysql.com/doc/refman/8.3/en/numeric-type-syntax.html
> The UNSIGNED attribute is deprecated for columns of type [FLOAT](https://dev.mysql.com/doc/refman/8.3/en/floating-point-types.html) (and any synonyms) and you should expect support for it to be removed in a future version of MySQL.

Although it's a separate question, do you think we need to warn from our side, @yaooqinn ?

Reply:
Hi @dongjoon-hyun. Yes, it's safe, we get `isSigned` from the JDBC standard API. It's OK for the DB side to remove this.

Reply:
> do you think we need to warn from our side

Oh, I missed this. This won't be necessary, it looks better to be warned when users use DDLs against MySQL DB. In this part, we actually act as a reader